toolkit/components/uniffi-js/ScaffoldingCall.h

```diff
@@ -0,0 +1,263 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim:set ts=2 sw=2 sts=2 et cindent: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_ScaffoldingCall_h
+#define mozilla_dom_ScaffoldingCall_h
+
+#include <tuple>
+#include <type_traits>
+#include "nsIGlobalObject.h"
+#include "nsPrintfCString.h"
+#include "mozilla/MozPromise.h"
+#include "mozilla/ResultVariant.h"
+#include "mozilla/dom/OwnedRustBuffer.h"
+#include "mozilla/dom/Promise.h"
+#include "mozilla/dom/ScaffoldingConverter.h"
+#include "mozilla/dom/UniFFIBinding.h"
+#include "mozilla/dom/UniFFIRust.h"
+
+namespace mozilla::dom {
```

Comment:
mildly surprised to not see a trailing `::uniffi` here (but that's probably just because I have no idea about this :) But assuming that's intentional/necessary, it probably kills my idea of dropping `dom` from the namespace 🤷

Reply:
I actually started doing that, but decided against it when I read this from the [Firefox C++ style guide](https://firefox-source-docs.mozilla.org/code-quality/coding-style/coding_style_cpp.html#c-namespaces): "Modules should avoid adding nested namespaces under mozilla, unless they are meant to contain names which have a high probability of colliding with other names in the code base. For example, Point, Path, etc."